app/models/metric/ci_mixin/capture.rb

```diff
@@ -46,20 +46,28 @@ def perf_capture_queue(interval_name, options = {})
 
     log_target = "#{self.class.name} name: [#{name}], id: [#{id}]"
 
+    # Determine the items to queue up
+    # cb is the task used to group cluster realtime metrics
     cb = nil
+    items = []
     if interval_name == 'historical'
       start_time = Metric::Capture.historical_start_time if start_time.nil?
-      end_time = Time.now.utc if end_time.nil?
+      end_time ||= 1.day.from_now.utc.beginning_of_day # Ensure no more than one historical collection is queue up in the same day
+      items = split_capture_intervals(interval_name, start_time, end_time)
     else
-      start_time = last_perf_capture_on unless start_time
-      end_time   = Time.now.utc unless end_time
+      # if last_perf_capture_on is earlier than 4.hour.ago.beginning_of_day,
+      # then create *one* realtime capture for start_time = 4.hours.ago.beginning_of_day (no end_time)
+      # and create historical captures for each day from last_perf_capture_on until 4.hours.ago.beginning_of_day
+      realtime_cut_off = 4.hours.ago.utc.beginning_of_day
+      if last_perf_capture_on && last_perf_capture_on < realtime_cut_off
+        items = split_capture_intervals("historical", last_perf_capture_on, realtime_cut_off)
+      end
+      # push realtime item to the front of the array
+      items.unshift([interval_name, realtime_cut_off])
```

Comment:
In the case of VMware this will be fine, I think.  Since the max we can get from the provider is `4.hours` of data.  However, for Hawkular, I think this breaks, since the max we can get from the provider is `7.days`.

So, without a start date, I think it will try to get the maximum amount of data available.  But, I'm not positive.